piptools/cache.py

```diff
@@ -159,6 +160,10 @@ def _reverse_dependencies(self, cache_keys):
         """
         # First, collect all the dependencies into a sequence of (parent, child) tuples, like [('flake8', 'pep8'),
         # ('flake8', 'mccabe'), ...]
-        return lookup_table((key_from_req(Requirement.parse(dep_name)), name)
+        return lookup_table((key_from_req(parse(dep_name)), name)
```

Comment:
I believe this is not related to the issue you are trying to fix. Please open another PR for the package extras related modification, and remove this commit.